From 5bd19d51d926a18acb7f447a7cc4d98ac0e0c419 Mon Sep 17 00:00:00 2001
From: Haikun Wang <Haikun.Wang@freescale.com>
Date: Mon, 11 May 2015 20:02:51 +0800
Subject: [PATCH 054/452] mtd: spi-nor: Change the timeout value of waiting
 flash ready

In kernel 3.16, we check the flash whether is ready before sending command
to it. But we check this after sending command to it in 3.19.
The max timeout value is 40s now and according the datasheet of DSPI flash on
ls2rdb its timeout value of eraseing whole chip is 480s.
So due to above reasons, we often get an timeout error when erase the whole
chip using kernel 3.19.
This patch increase the timeout value to 480s.

Signed-off-by: Haikun Wang <haikun.wang@freescale.com>
Change-Id: I56bb4041a9f6dc20055859e4cc3c80b64171b655
Reviewed-on: http://git.am.freescale.net:8181/36167
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Stuart Yoder <stuart.yoder@freescale.com>
(cherry picked from commit 619b52c89847474e2f5aeea1cc2142ebbc2443cf)
[Xulin: Original patch taken from FSL LS2085 SDK EAR6.0,
LS2085A-SDK-SOURCE-20160304-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/mtd/spi-nor/spi-nor.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/mtd/spi-nor/spi-nor.c b/drivers/mtd/spi-nor/spi-nor.c
index b61b6fa..a8c41a8 100644
--- a/drivers/mtd/spi-nor/spi-nor.c
+++ b/drivers/mtd/spi-nor/spi-nor.c
@@ -24,7 +24,7 @@
 #include <linux/mtd/spi-nor.h>
 
 /* Define max times to check status register before we give up. */
-#define	MAX_READY_WAIT_JIFFIES	(40 * HZ) /* M25P16 specs 40s max chip erase */
+#define	MAX_READY_WAIT_JIFFIES	(480 * HZ) /* M25P16 specs 40s max chip erase */
 
 #define SPI_NOR_MAX_ID_LEN	6
 
-- 
1.7.5.4

